### PR TITLE
{chem}[foss/2016b] QuantumESPRESSO 5.4.0

### PR DESCRIPTION
**Pull request by LiangFr**
QE Easyconfig file should be used together with [Easyblock file in PR #954](https://github.com/hpcugent/easybuild-easyblocks/pull/954) @ocaisa

### CODE REVIEW

File: easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-5.4.0-foss-2016b-hybrid.eb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    'http://www.qe-forge.org/gf/download/frsrelease/211/960/', # pwcond-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/958/', # xspectra-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/969/', # EPW-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/963/', # test-suite-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/957/', # GWW-5.4.0.tar.gz
+]
+
+# Hybrid option exist only for backward compatibility, use toolchain option openmp instead.
+# hybrid = True
+# Patch file to avoil the CPP problem generated by different OSes
```
Comment (ocaisa) [inline easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-5.4.0-foss-2016b-hybrid.eb]: avoid

File: easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-5.4.0-foss-2016b-hybrid.eb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    'http://www.qe-forge.org/gf/download/frsrelease/211/959/', # neb-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/962/', # PHonon-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/960/', # pwcond-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/958/', # xspectra-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/969/', # EPW-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/963/', # test-suite-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/957/', # GWW-5.4.0.tar.gz
+]
+
+# Hybrid option exist only for backward compatibility, use toolchain option openmp instead.
```
Comment (ocaisa) [inline easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-5.4.0-foss-2016b-hybrid.eb]: exists

**Comment by boegel** [comment]
@LiangFr problem detected by Travis:

```
AssertionError: Filename '.../easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-5.4.0-foss-2016b-hybrid.eb'
of parsed easyconfig matches expected filename 'QuantumESPRESSO-5.4.0-foss-2016b.eb'
```

seems like you're missing this in the QE easyconfig (right below `version =`):

```
versionsuffix = '-hybrid'
```

or rename the easyconfig file to remove the `-hybrid` part

**Comment by LiangFr** [comment]
@boegel Thanks. I activated the 'hybrid= True' option in .eb file. But I think it will just be used to pass the Travis CI test with '-hybrid' in file title. Because inside the easyconfig file, we've already put `toolchainopts = {'usempi': True, 'openmp': True}`  with which our easyblock file in PR#953 is adapted.  @ocaisa

**Comment by boegel** [comment]
@LiangFr the `hybrid` option does not have an effect on how the easyconfig file is named, the `versionsuffix = '-hybrid'` *does* however, so you should either add that line to the easyconfig file, or removing the `-hybrid` from the filename...

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2153.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/a38cd02c0b79f5ad33ab0ef6d8b194b8 for a full test report.

**Comment by ocaisa** [comment]
Did you use the new easyblock?

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2498.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/032c720124fe6d6e5e5da237478c2bfe for a full test report.

**Comment by boegel** [comment]
@ocaisa ah, no... Will retest, sorry.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2631.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/191c4a2603fe7e51bd6d0f92005fb189 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/46da8bccfa7dc83b511345ca7b897988 for a full test report.

File: easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-5.4.0-foss-2016b-hybrid.eb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    'http://www.qe-forge.org/gf/download/frsrelease/211/958/', # xspectra-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/969/', # EPW-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/963/', # test-suite-5.4.0.tar.gz
+    'http://www.qe-forge.org/gf/download/frsrelease/211/957/', # GWW-5.4.0.tar.gz
+]
+
+# Hybrid option exists only for backward compatibility, use toolchain option openmp instead.
+hybrid = True
+# Patch file to avoid the CPP problem generated by different OSes
+patches = ['QE_gfortran_f90rule.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-5.4.0-foss-2016b-hybrid.eb]: please rename to something like `QuantumESPRESSO-5.4.0_fix-f90rule-gfortran.patch`

File: easybuild/easyconfigs/q/QuantumESPRESSO/QE_gfortran_f90rule.patch [hunk 4]
```diff
@@ -0,0 +1,12 @@
+diff -rupN espresso-5.4.0.orig/install/make.sys.in espresso-5.4.0/install/make.sys.in
```
Comment (boegel) [inline easybuild/easyconfigs/q/QuantumESPRESSO/QE_gfortran_f90rule.patch]: @LiangFr can you include a short description of what the patch does above this line (anything above is considered as comments), and include yourself as an author?

**Comment by boegel** [comment]
@LiangFr other than the final remarks about the patch file, this should be good to go, thanks!

cc @ocaisa

File: easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-5.4.0-foss-2016b-hybrid.eb (hunk 3 above)
Comment (LiangFr) [inline easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-5.4.0-foss-2016b-hybrid.eb]: Ok.

**Comment by boegel** [comment]
lgtm`

**Comment by boegel** [comment]
Going in, thanks @LiangFr!

**Pull request merged**